### DISCUSSION TITLE
Favicons with MetaData API files system not working

### DISCUSSION THREAD

**Discussion by da-kicks-87** [amannn/next-intl#1519] [Q&A]
### Description

This is for   Next.js 14.2.16. -  App Router

Hello after installing next-intel the favicon no longer load from the file-based metadata system. I had tried to have the favion.ico in `/src/app/[locale]` and `/src/app`. (https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons)

I got the favicons to work by doing the alternative by using Next.js metadata export object and putting the favicons in the `/public/favicons`

Eg from layout.js:  

```
export const metadata = {  icons: {
    icon: [
      { url: '/favicons/favicon-96x96.png', type:'image/png', sizes: '96x96' },
      { url: '/favicons/favicon.svg', type:'image/svg+xml',},
      { url: '/favicons/favicon.ico' },
    ],
    shortcut: '/favicons/favicon.ico',
    apple: { url: '/favicons/apple-touch-icon.png', sizes: '180x180' },
    other: {
      rel: 'manifest',
      url: '/favicons/site.webmanifest'
    }
  }, }
```

However the Next.js guide mention i should be using the special metadata files inside the app folder for favicons.
What should I do?

### Verifications

- [X] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on [GitHub](https://github.com/amannn/next-intl/issues) and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Mandatory reproduction URL

/sorry-offline

### Reproduction description

See above description

### Expected behaviour

See above description

**Comment by amannn** [comment]
The `next-intl` examples currently simply put a [`favicon.ico` in `public`](https://github.com/amannn/next-intl/tree/main/examples/example-app-router/public)—that seems to work for me. I'm not aware of if there are implications for other patterns. There might be details relevant to your app, like a potential `matcher` you're using the middleware config.

I'll move this to a discussion since it's a Next.js usage question and not a bug report about `next-intl`.

**Comment by da-kicks-87** [comment]
My middleware.js is like this:

```
import createMiddleware from 'next-intl/middleware';
import {routing} from './i18n/routing';
 
export default createMiddleware(routing);
 
export const config = {
  // Match only internationalized pathnames
  matcher: ['/', '/(fr|en)/:path*', '/((?!_next|_vercel|.*\\..*).*)']
};
```

**Comment by FireLord** [comment]
Same here. I have added in /public/favicon.ico and even in src/app/favicon.ico still it’s not working. Even the example one when ran locally doesn’t load favicon.

**Reply by FireLord** [reply]
@da-kicks-87 [this](https://stackoverflow.com/questions/61836949/why-is-my-favicon-not-working-in-my-next-js-app) worked for me. added icon.ico in src/app/ dir

**Reply by da-kicks-87** [reply]
Doing that only works for the default .ico size. The other sizes don't get auto generated. The Next.js docs says they should, but they don't. Also I don't see how the webmainifest icons would work with the file-based metadata system.

I will go back to the Meta data object and explicitly define them there. It's the only way I can get them to render.
